jobs/gtfs-schedule-validator/gtfs_schedule_validator.py

```diff
@@ -0,0 +1,190 @@
+__version__ = "0.1.0"
+
+import gzip
+import json
+import logging
+import os
+import subprocess
+import tempfile
+import traceback
+from datetime import datetime
+from pathlib import Path
+from typing import Dict
+
+import pendulum
+import typer
+from calitp.storage import (
+    fetch_all_in_partition,
+    GTFSFeedExtractInfo,
+    get_fs,
+    GTFSScheduleFeedValidation,
+    GTFSFeedType,
+    JSONL_GZIP_EXTENSION,
+    GTFSScheduleFeedExtractValidationOutcome,
+    ScheduleValidationResult,
+)
+
+JAVA_EXECUTABLE_PATH_KEY = "GTFS_SCHEDULE_VALIDATOR_JAVA_EXECUTABLE"
+SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY = "GTFS_SCHEDULE_VALIDATOR_JAR"
+JAR_DEFAULT = typer.Option(
+    default=os.environ.get(SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY),
+    help="Path to the GTFS Schedule Validator JAR",
+)
+
+app = typer.Typer()
+logging.basicConfig()
+
+
+def execute_schedule_validator(
+    fs,
+    zip_path: Path,
+    output_dir: Path,
+    jar_path: Path = os.environ.get(SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY),
+    verbose=False,
+) -> (Dict, Dict):
+    if not isinstance(zip_path, Path):
+        raise TypeError("must provide a path to the zip file")
+
+    args = [
+        os.getenv(JAVA_EXECUTABLE_PATH_KEY, default="java"),
+        "-jar",
+        str(jar_path),
+        "--input",
+        str(zip_path),
+        "--output_base",
+        str(output_dir),
+        "--feed_name",
+        "us-na",
+    ]
+
+    report_path = Path(output_dir) / "report.json"
+    system_errors_path = Path(output_dir) / "system_errors.json"
+
+    typer.secho(f"executing schedule validator: {' '.join(args)}")
+    subprocess.run(
+        args,
+        capture_output=True,
+    ).check_returncode()
+
+    with open(report_path) as f:
+        report = json.load(f)
+
+    with open(system_errors_path) as f:
+        system_errors = json.load(f)
+
+    return report, system_errors
+
+
+@app.command()
+def validate_extract(
+    zip_path: Path,
+    output_dir: Path,
+    jar_path: Path = os.environ.get(SCHEDULE_VALIDATOR_JAR_LOCATION_ENV_KEY),
+    verbose=False,
+) -> None:
+    """"""
+    execute_schedule_validator(
+        fs=get_fs(),
+        zip_path=zip_path,
+        output_dir=output_dir,
+        jar_path=jar_path,
+        verbose=verbose,
+    )
+
+
+@app.command()
+def validate_day(
+    day: datetime = typer.Argument(
+        default=pendulum.today(),
```

Comment:
Do we have to define a default for this? I guess since it gets them based on partition it should be ok, but I am wondering about weirdness with DAG execution times... Kind of would prefer that this has to be explicitly given.

Reply:
yeah that's fair, I was mostly facilitating using the command directly via CLI but I'm not really convinced it's a great pattern yet... maybe we should be defining the actual typer entrypoints in `__main__.py`?

Reply:
Kind of defer to you on that side, or maybe we can have a commented version for testing? Agree it's useful to leave testing breadcrumbs for local dev but just worry about leaving them in unmarked